controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -274,3 +318,39 @@ func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1beta
 		"snapshot.Name", snapshot.Name)
 	return pipelineRun, nil
 }
+
+func (a *Adapter) removeFinalizerAndContinueProcessing() (controller.OperationResult, error) {
+	// Remove before merge
+	a.logger.Info(fmt.Sprintf("Removing finalizer '%s' from build pipeline run '%s'", finalizer, a.pipelineRun.Name))
+
+	if !h.HasPipelineRunFinished(a.pipelineRun) {
+		// Remove before merge
+		a.logger.Info("build PLR has not finished")
+		return controller.ContinueProcessing()
+	}
+	if !controllerutil.ContainsFinalizer(a.pipelineRun, finalizer) {
+		// Remove before merge
+		a.logger.Info("build PLR does not contain finalizer")
+		return controller.ContinueProcessing()
+	}
+
+	err := retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		patch := client.MergeFrom(a.pipelineRun.DeepCopy())
+		if !controllerutil.RemoveFinalizer(a.pipelineRun, finalizer) {
```

Comment:
Now that STONEINTG-599 was merged, we can probably use [RemoveFinalizerFromPipelineRun](https://github.com/redhat-appstudio/integration-service/blob/main/helpers/integration.go#L411-L425)() here for commonality.